sktime/classification/deep_learning/rnn.py

```diff
@@ -66,7 +67,16 @@ def __init__(
     ):
         _check_dl_dependencies(severity="error")
         super().__init__()
-        self.num_epochs = num_epochs
+        # Deprecated Parameter Handling
+        if num_epochs is not None:
+            warnings.warn(
+                "The parameter 'num_epochs' is deprecated and will be removed in version 0.26.0. "
+                "Use 'n_epochs' instead. To avoid this warning, update your code to use 'n_epochs'.",
+                DeprecationWarning, stacklevel=2
+            )
+            self.n_epochs = num_epochs
```

Comment:
this violates sklearn interface, all init params should be written to self with the same name.

That is, you need to write self.num_epochs as wll as n_epochs